### PR TITLE
Added filter for author/creator

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | fixes #issuenum |
| Related issues/PRs | #issuenum |
| License | MIT |
| Documentation PR | sulu/sulu-docs#prnum |
#### What's in this PR?

New feature for article list view: Added filter for author/creator.
#### To Do
- Elasticsearch: Add new fields (authors_id, creator_contact_id)
- Overlay-Filter: Add 
- Dropdown: Three items (Alle, Nur meine, Nur von...)
- Overlay-Filter: datagrid with search
- Filter by authors_id && creator_contact_id

### CODE REVIEW

**Comment by trickreich** [comment]
@wachterjohannes In my opinion this would be nice feature und should be included at least in version 1.0

**Comment by wachterjohannes** [comment]
i would say we do it as soon as possible eventually for 0.3.0? what do you think?

File: Document/ArticleViewDocumentInterface.php [hunk 1]
```diff
@@ -351,4 +351,40 @@ public function getLocalizationState();
      * @return $this
      */
     public function setLocalizationState(LocalizationStateViewObject $localizationState);
+
+    /**
+     * @param string $authorId
```
Comment (wachterjohannes) [inline Document/ArticleViewDocumentInterface.php]: add descriptions to this function like the others in this class.

File: Resources/public/js/components/articles/list/main.js [hunk 2]
```diff
@@ -335,6 +328,73 @@ define([
...
+                        dropdownItems: [
+                            {
+                                type: 'columnOptions'
+                            }
+                        ]
+                    }
+                }
+            });
+
+            listToolbarTemplate.push({
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/list/main.js]: why not retrieving this button also from `buttons.get`?

```
contactIdFilter: {
    options: {
        icon: 'filter',
        title: this.translations.filterAll,
        group: 2,
        ...
    }
}
```

File: Resources/public/js/components/articles/list/main.js [hunk 3]
```diff
@@ -335,6 +328,73 @@ define([
...
+         * @param item {Object}
+         */
+        applyFilterToList: function(item) {
+            var contactId = null;
+
+            if (!!item.id) {
+                switch(item.id) {
+                    case 'me':
+                        contactId = this.sandbox.sulu.user.id;
+
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/list/main.js]: no new line

File: Resources/public/js/components/articles/list/main.js [hunk 4]
```diff
@@ -335,6 +328,73 @@ define([
...
+                        break;
+                    default:
+                        contactId = null;
+
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/list/main.js]: no new line

File: Resources/translations/sulu/backend.en.xlf [hunk 5]
```diff
@@ -118,6 +118,18 @@
                 <source>sulu_article.smart-content.author-full-name</source>
                 <target>Author (alphabetic)</target>
             </trans-unit>
+            <trans-unit id="list-filter-01" resname="sulu_article.list.filter.by">
+                <source>sulu_article.list.filter.by</source>
+                <target>Only from...</target>
```
Comment (wachterjohannes) [inline Resources/translations/sulu/backend.en.xlf]: could this be removed?

**Pull request merged**